examples/births-deaths/infection_manager.rs

```diff
@@ -0,0 +1,178 @@
+        .unwrap_or(&HashSet::<plan::Id>::new())
+        .clone();
+
+    for plan_id in plans_set {
+        context.cancel_plan(&plan_id);
+    }
+
+    remove_recovery_plan_data(context, person_id);
+}
+
+fn handle_infection_status_change(
+    context: &mut Context,
+    event: PersonPropertyChangeEvent<InfectionStatusType>,
+) {
+    if matches!(event.current, InfectionStatus::I) {
+        schedule_recovery(context, event.person_id);
+    }
+    if matches!(event.current, InfectionStatus::R) {
+        remove_recovery_plan_data(context, event.person_id);
+    }
+}
+
+fn handle_person_removal(context: &mut Context, event: PersonPropertyChangeEvent<Alive>) {
+    if !event.current {
+        cancel_recovery_plans(context, event.person_id);
+    }
+}
+pub fn init(context: &mut Context) {
+    context.subscribe_to_event(
+        move |context, event: PersonPropertyChangeEvent<InfectionStatusType>| {
+            handle_infection_status_change(context, event);
+        },
+    );
+
+    context.subscribe_to_event(move |context, event: PersonPropertyChangeEvent<Alive>| {
+        handle_person_removal(context, event);
+    });
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::population_manager::ContextPopulationExt;
+    use ixa::context::Context;
+    use ixa::define_data_plugin;
+    use ixa::global_properties::ContextGlobalPropertiesExt;
+    use ixa::people::{ContextPeopleExt, PersonPropertyChangeEvent};
+    use ixa::random::ContextRandomExt;
+
+    use crate::parameters_loader::{FoiAgeGroups, ParametersValues};
+    define_data_plugin!(RecoveryPlugin, usize, 0);
+    define_data_plugin!(PlansPlugin, usize, 0);
+
+    #[test]
+    fn test_handle_infection_change_with_deaths() {
+        let p_values = ParametersValues {
+            population: 10,
+            max_time: 10.0,
+            seed: 42,
+            birth_rate: 0.0,
+            death_rate: 0.1,
+            foi_groups: Vec::<FoiAgeGroups>::new(),
+            infection_duration: 5.0,
+            output_file: ".".to_string(),
+            demographic_output_file: ".".to_string(),
+        };
+
+        let mut context = Context::new();
+
+        context.set_global_property_value(Parameters, p_values.clone());
+        context.init_random(p_values.seed);
+        init(&mut context);
+
+        context.subscribe_to_event(
+            move |context, event: PersonPropertyChangeEvent<InfectionStatusType>| {
+                if matches!(event.current, InfectionStatus::R) {
+                    *context.get_data_container_mut(RecoveryPlugin) += 1;
+                }
+            },
+        );
+
+        let population_size: usize = 10;
+        for _ in 0..population_size {
+            let person = context.create_new_person(0.0);
+
+            context.add_plan(1.0, move |context| {
+                context.set_person_property(person, InfectionStatusType, InfectionStatus::I);
+            });
+        }
+
+        context.add_plan(1.1, move |context| {
```

Comment:
why 1.1?

Reply:
I wanted to kill right after infection so that recovery would have to be canceled. It could also be at 1.0, I guess.